pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -63,21 +96,52 @@ func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
 	if !current.CanTransitionTo(next) {
 		return fmt.Errorf("invalid state transition: %s -> %s", current.Name(), next.Name())
 	}
-	// TODO: call pre-transition listeners -  Issue: https://github.com/hyperledger/aries-framework-go/issues/140
-	followup, err := next.Execute(msg)
+
+	// trigger the event if the message type is registered (if it's not outbound)
+	eventsTriggered := false
+	if !msg.Outbound {
+		eventsTriggered, err = s.sendEvent(msg, thid, next)
+		if err != nil {
+			return fmt.Errorf("send events failed: %w", err)
+		}
+	}
+
+	// if no events are trigger continue the execution
+	if !eventsTriggered {
+		return s.handle(&message{Msg: msg, ThreadID: thid, NextStateName: next.Name()})
+	}
+
+	return nil
+}
+
+// Register events on DID Exchange protocol. The events are triggered for incoming ConnectionRequest, ConnectionResponse
```

Comment:
There is an undocumented side-effect: auto execution when you don't register; auto execution if you do.

Also as a TODO: you might only be registering for events not for callbacks. Related to the thought above: https://github.com/hyperledger/aries-framework-go/pull/240/files#r324795675

Reply:
Updated the Godoc and created a TODO. #261